x/claim/keeper/hooks.go

```diff
@@ -12,48 +15,80 @@ import (
 	"github.com/tharsis/evmos/x/claim/types"
 )
 
-// TODO: add hooks for IBC
+var (
+	_ transfertypes.ICS4Wrapper = Hooks{}
+	_ evmtypes.EvmHooks         = Hooks{}
+	_ govtypes.GovHooks         = Hooks{}
+	_ stakingtypes.StakingHooks = Hooks{}
+)
+
+// Hooks wrapper struct for the claim keeper
+type Hooks struct {
+	k Keeper
+}
+
+// Return the wrapper struct
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k}
+}
 
 func (k Keeper) AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress) {
-	_, err := k.ClaimCoinsForAction(ctx, voterAddr, types.ActionVote)
+	claimRecord, found := k.GetClaimRecord(ctx, voterAddr)
+	if !found {
+		// no-op
+		return
+	}
+
+	params := k.GetParams(ctx)
+
+	_, err := k.ClaimCoinsForAction(ctx, voterAddr, claimRecord, types.ActionVote, params)
 	if err != nil {
 		panic(err.Error())
 	}
 }
 
 func (k Keeper) AfterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) {
-	_, err := k.ClaimCoinsForAction(ctx, delAddr, types.ActionDelegate)
+	claimRecord, found := k.GetClaimRecord(ctx, delAddr)
+	if !found {
+		// no-op
+		return
+	}
+
+	params := k.GetParams(ctx)
+
+	_, err := k.ClaimCoinsForAction(ctx, delAddr, claimRecord, types.ActionDelegate, params)
 	if err != nil {
 		panic(err.Error())
 	}
 }
 
 func (k Keeper) AfterEVMStateTransition(ctx sdk.Context, from common.Address, to *common.Address, receipt *ethtypes.Receipt) error {
 	fromAddr := sdk.AccAddress(from.Bytes())
-	_, err := k.ClaimCoinsForAction(ctx, fromAddr, types.ActionEVM)
+
+	claimRecord, found := k.GetClaimRecord(ctx, fromAddr)
+	if !found {
+		// no-op
+		return nil
+	}
+
+	params := k.GetParams(ctx)
+
+	_, err := k.ClaimCoinsForAction(ctx, fromAddr, claimRecord, types.ActionEVM, params)
 	if err != nil {
 		return err
 	}
+
 	return nil
 }
 
 // ________________________________________________________________________________________
 
-// Hooks wrapper struct for the claim keeper
-type Hooks struct {
-	k Keeper
-}
+// IBC callbacks and transfer handlers
 
-// TODO: add IBC hooks
-var (
-	_ evmtypes.EvmHooks         = Hooks{}
-	_ govtypes.GovHooks         = Hooks{}
-	_ stakingtypes.StakingHooks = Hooks{}
-)
-
-// Return the wrapper struct
-func (k Keeper) Hooks() Hooks {
-	return Hooks{k}
+// SendPacket implements the ICS4Wrapper interface from the transfer module.
+// It calls the underlying SendPacket function directly to move down the middleware stack.
+func (h Hooks) SendPacket(ctx sdk.Context, channelCap *capabilitytypes.Capability, packet exported.PacketI) error {
+	return h.k.ics4Wrapper.SendPacket(ctx, channelCap, packet)
```

Comment:
should the `ics4Wrapper `be the transfer keeper in `app.go`?

Reply:
No `ics4Wrapper` for hooks should be channel keeper. 

transfer -> hooks -> channel keeper
ics4wrapper in transfer is hooks
ics4wrapper in hooks is channel keeper